lib/_emerge/depgraph.py

```diff
@@ -1942,7 +1944,9 @@ def _process_slot_conflicts(self):
         for conflict in self._dynamic_config._package_tracker.slot_conflicts():
             self._process_slot_conflict(conflict)
 
-        if self._dynamic_config._allow_backtracking:
+        if self._dynamic_config._allow_backtracking and not any(
+            opt in self._frozen_config.myopts for opt in _NO_MERGE_OPTS
+        ):
             self._slot_operator_trigger_reinstalls()
```

Comment:
It seems like --buildpkgonly needs to be in a class of its own here, since --fetch* should behave mostly like regular build commands in order to avoid prevent discrepancies in behavior (like the slot conflicts in https://bugs.gentoo.org/161422 thought those would exist without --fetchonly you get the idea).